### ISSUE TITLE
Implement SQL window functions for multi-shard queries that are not pushdownable

### ISSUE THREAD

**Issue by marcocitus** [citusdata/citus#1847]
Window functions in queries across shards currently error out:

```sql
SELECT row_number() OVER() FROM test;
ERROR:  could not run distributed query with window functions
HINT:  Consider using an equality filter on the distributed table's partition column.
```

Except when using a partition column filter:
```sql
SELECT row_number() OVER() FROM test WHERE x = 1;
 row_number 
------------
          1
          2
(2 rows)
```

or when using a partition by clause:
```sql
SELECT row_number() OVER(PARTITION BY x) FROM test;
 row_number 
────────────
          1
          2
(2 rows)
```

**Comment by furkansahin** [comment]
I think this issue is resolved when #2008 is implemented. I'm closing the issue.

**Comment by marcocitus** [comment]
Still gives me an error:
```
postgres=# SELECT row_number() OVER() FROM test;
ERROR:  could not run distributed query because the window function that is used cannot be pushed down
HINT:  Window functions are supported in two ways. Either add an equality filter on the distributed tables' partition column or use the window functions with a PARTITION BY clause containing the distribution column
```
